src/Sulu/Bundle/SearchBundle/EventSubscriber/BlameTimestampSubscriber.php

```diff
@@ -0,0 +1,201 @@
+        );
+    }
+
+    /**
+     * Map blame and timestamp information to the search document
+     *
+     * @param PreIndexEvent $event
+     */
+    public function handleBlameTimestamp(PreIndexEvent $event)
+    {
+        $subject = $event->getSubject();
+        $document = $event->getDocument();
+
+        if ($subject instanceof UserBlameInterface) {
+            $this->mapCreatorAndChanger($document, $subject->getCreator(), $subject->getChanger());
+        }
+
+        if ($subject instanceof TimestampableInterface) {
+            $this->mapTimestamp($document, $subject->getCreated(), $subject->getChanged());
+        }
+
+        if ($subject instanceof StructureInterface) {
+            $this->handleStructure($subject, $document);
+        }
+    }
+
+    /**
+     * Map the changer and created from the field data to
+     * the search document (we don't include the field data in the search result API)
+     *
+     * @param HitEvent $event
+     */
+    public function handleBlameTimestampHitMapping(HitEvent $event)
+    {
+        $document = $event->getHit()->getDocument();
+        $this->doHandleBlameTimestampHitMapping($document);
+    }
+
+    /**
+     * @param Document $document
+     */
+    private function doHandleBlameTimestampHitMapping(Document $document)
+    {
+        $document->setCreatorName($this->getFieldValue($document, 'creator'));
+        $document->setChangerName($this->getFieldValue($document, 'changer'));
+        $document->setCreatorId($this->getFieldValue($document, 'creator_id'));
+        $document->setChangerId($this->getFieldValue($document, 'changer_id'));
+        $document->setCreated($this->getFieldValue($document, 'created'));
+        $document->setChanged($this->getFieldValue($document, 'changed'));
+    }
+
+    /**
+     * Return the named field from the document or return null
+     *
+     * @param mixed $document
+     * @param mixed $fieldName
+     */
+    private function getFieldValue($document, $fieldName)
+    {
+        if (false === $document->hasField($fieldName)) {
+            return null;
+        }
+
+        return $document->getField($fieldName)->getValue();
+    }
+
+    /**
+     * Handle Sulu Structure objects
+     *
+     * @param StructureInterface $structure
+     * @param Document $document
+     */
+    private function handleStructure(StructureInterface $structure, Document $document)
+    {
+        $creatorId = $structure->getCreator();
+        $changerId = $structure->getChanger();
+        $creator = null;
+        $changer = null;
+
+        // fo not try to do this if the Sulu Security component is not registered
+        // for some reason.
+        //
+        // TODO: ->find does not seem to work when using the UserInterface, so we
+        //       are now coupled to the Sulu User entity.
+        if (class_exists(User::class)) {
+            if ($creatorId) {
+                $creator = $this->entityManager->find(User::class, $creatorId);
```

Comment:
I think @Ocramius told me last tuesday that it will work in the next version of doctrine.